gemfire-core/src/main/java/com/gemstone/gemfire/internal/cache/AbstractLRURegionMap.java

```diff
@@ -680,6 +683,11 @@ public final long centralizedLruUpdateCallback(boolean includeOffHeapBytes) {
             }
           }
           int evicted = evictEntry(removalEntry, stats);
+          if (skipLockedEntries) {
+            // release the lock held on by getLRUEntry
+            UnsafeHolder.getUnsafe().monitorExit(removalEntry);
```

Comment:
monitorExit is deprecated. We might have to soon change this perhaps.

Reply:
As mentioned in the email exchange on this, Unsafe monitor methods have been removed from JDK 9. We need a flag or something on RegionEntry that every piece of code acquiring the monitor on RegionEntry must first set. However, the atomicity of that can still be a problem so another option could be to use the transactional EX lock where one can use tryLock like above.